tests/device_tests/test_msg_ethereum_signtx_eip1559.py

```diff
@@ -0,0 +1,385 @@
+            eip2718_type=2,
+            access_list=[
+                messages.EthereumAccessList(
+                    address="0xde0b295669a9fd93d5f28d9ec85e40f4cb697bae",
+                    storage_keys=[
+                        bytes.fromhex(
+                            "0000000000000000000000000000000000000000000000000000000000000003"
+                        ),
+                        bytes.fromhex(
+                            "0000000000000000000000000000000000000000000000000000000000000007"
+                        ),
+                    ],
+                ),
+                messages.EthereumAccessList(
+                    address="0xbb9bc244d798123fde783fcc1c72d3bb8c189413",
+                    storage_keys=[
+                        bytes.fromhex(
+                            "0000000000000000000000000000000000000000000000000000000000000006"
+                        ),
+                        bytes.fromhex(
+                            "0000000000000000000000000000000000000000000000000000000000000007"
+                        ),
+                        bytes.fromhex(
+                            "0000000000000000000000000000000000000000000000000000000000000009"
+                        ),
+                    ],
+                ),
+            ],
+        )
+
+    assert sig_v == 1
+    assert (
+        sig_r.hex()
+        == "718a3a30827c979975c846d2f60495310c4959ee3adce2d89e0211785725465c"
+    )
+    assert (
+        sig_s.hex()
+        == "7d0ea2a28ef5702ca763c1f340427c0020292ffcbb4553dd1c8ea8e2b9126dbc"
+    )
+
+
+def test_ethereum_signtx_known_erc20_token(client):
+    with client:
+
+        data = bytearray()
+        # method id signalizing `transfer(address _to, uint256 _value)` function
+        data.extend(bytes.fromhex("a9059cbb"))
+        # 1st function argument (to - the receiver)
+        data.extend(
+            bytes.fromhex(
+                "000000000000000000000000574bbb36871ba6b78e27f4b4dcfb76ea0091880b"
+            )
+        )
+        # 2nd function argument (value - amount to be transferred)
+        data.extend(
+            bytes.fromhex(
+                "000000000000000000000000000000000000000000000000000000000bebc200"
+            )
+        )
+        # 200 000 000 in dec, decimals of ADT = 9, trezor1 displays 0.2 ADT, Trezor T 200 000 000 Wei ADT
+
+        sig_v, sig_r, sig_s = ethereum.sign_tx(
+            client,
+            n=parse_path("44'/60'/0'/0/0"),
+            nonce=0,
+            max_gas_fee=20,
+            max_priority_fee=1,
+            gas_limit=20,
+            gas_price=None,
+            # ADT token address
+            to="0xd0d6d6c5fe4a677d343cc433536bb717bae167dd",
+            chain_id=1,
+            # value needs to be 0, token value is set in the contract (data)
+            value=0,
+            data=data,
+            eip2718_type=2,
+        )
+
+    # taken from T1 might not be 100% correct but still better than nothing
```

Comment:
this comment is probably incorrect :) dtto below

Reply:
Yep - fixed!